src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -1939,3 +1939,300 @@ def test_sql_subscription_usages(self):
+        database_name = "createZonedDb"
+        database_name_2 = "createUnzonedDb"
+        database_name_3 = "createUnzonedDbNoParam"
+
+        rg = resource_group
+        loc_display = "East US 2"
+
+        # Test creating database with zone resilience set to true.  Expect zone resilient database created.
+        self.cmd('sql db create -g {} --server {} --name {} --edition {} --zone-redundant {}'
+                 .format(rg, server, database_name, "Premium", True),
+                 checks=[
+                     JMESPathCheck('resourceGroup', rg),
+                     JMESPathCheck('name', database_name),
+                     JMESPathCheck('location', loc_display),
+                     JMESPathCheck('elasticPoolName', None),
+                     JMESPathCheck('edition', 'Premium'),
+                     JMESPathCheck('zoneRedundant', True)]).get_output_in_json()
+
+        # Test creating database with zone resilience set to false.  Expect regular database created.
+        self.cmd('sql db create -g {} --server {} --name {} --edition {} --zone-redundant {}'
+                 .format(rg, server, database_name_2, "Premium", False),
+                 checks=[
+                     JMESPathCheck('resourceGroup', rg),
+                     JMESPathCheck('name', database_name_2),
+                     JMESPathCheck('location', loc_display),
+                     JMESPathCheck('elasticPoolName', None),
+                     JMESPathCheck('edition', 'Premium'),
+                     JMESPathCheck('zoneRedundant', False)]).get_output_in_json()
+
+        # Test creating database with no zone resilience set.  Expect regular database created.
+        self.cmd('sql db create -g {} --server {} --name {} --edition {}'
+                 .format(rg, server, database_name_3, "Premium"),
+                 checks=[
+                     JMESPathCheck('resourceGroup', rg),
+                     JMESPathCheck('name', database_name_3),
+                     JMESPathCheck('location', loc_display),
+                     JMESPathCheck('elasticPoolName', None),
+                     JMESPathCheck('status', 'Online'),
+                     JMESPathCheck('edition', 'Premium'),
+                     JMESPathCheck('zoneRedundant', False)]).get_output_in_json()
+
+    @ResourceGroupPreparer()
+    @SqlServerPreparer()
+    def test_sql_update_zone_resilient_database(self, resource_group, resource_group_location, server):
+        database_name = "updateToZonedDb"
+        database_name_2 = "updateToUnzonedDb"
+        database_name_3 = "updateNoParamForUnzonedDb"
+        database_name_4 = "updateNoParamForZonedDb"
+
+        rg = resource_group
+        loc_display = "East US 2"
+
+        # Test running update on regular database with zone resilience set to true.  Expect zone resilience to update to true.
+        self.cmd('sql db create -g {} --server {} --name {} --edition {}'
```

Comment:
This is already the 1st step in the "create zone redundant" test case, so you can combine the 2 test cases to reduce test e2e run time.